rfc-116-store-attachment-data-in-content-items.md

```diff
@@ -0,0 +1,321 @@
+   accordingly, with these fields:
+
+   - `base_attachment_asset`:
+     - `url` (required)
+     - `content_type` (required)
+     - `title`
+   - `attachment_asset` (extends `base_attachment_asset`):
+     - no extra fields at this time
+   - `specialist_publisher_attachment_asset` (extends `base_attachment_asset`):
+     - `content_id` (required)
+     - `created_at`
+     - `updated_at`
+   - `image_asset`:
+     - `url` (required)
+     - `content_type` (required)
+
+   **Migration concerns:** This would need to be done in three steps:
+   1. Update manuals-publisher to not set the `created_at` and `updated_at` for attachments.
+   2. Republish all manual sections.
+   3. Update the schemas.
+
+1. Add new optional fields for extra metadata:
+
+   - `base_attachment_asset`:
+     - `accessible`
+     - `alternative_format_contact_email`
+     - `filename`
+     - `file_size`
+     - `locale`
+     - `number_of_pages`
+     - `preview_url`
+   - `attachment_asset`
+     - `id`
+   - `image_asset`
+     - `alt_text`
+     - `caption`
+     - `credit`
+
+   **Migration concerns:** This can be done without breaking
+   compatibility, as all new fields are optional.
+
+1. Add new attachment type for publication attachments:
+
+    `publication_attachment_asset` (extends `attachment_asset`) with fields:
+    - `attachment_type` (required)
+    - `command_paper_number`
+    - `hoc_paper_number`
+    - `isbn`
+    - `parliamentary_session`
+    - `unique_reference`
+    - `unnumbered_command_paper`
+    - `unnumbered_hoc_paper`
+
+   I've left out the following fields because they can be deduced from
+   others: `csv?`, `external?`, `external_url`, `file_extension`, `html?`,
+   `opendocument?`, `pdf?`
+
+   And these fields because they don't seem to be used:
+   `print_meta_data_contact_address`, `web_isbn`.
+
+   And these fields because the Publishing Workflow team are planning
+   to remove them from Whitehall: `order_url`, `price_in_pence`.
+
+   And these fields because they don't need to be exposed outside the
+   publishing app: `attachable_id`, `attachment_data_id`, `deleted`,
+   `ordering`, `slug`.
+
+   **Migration concerns:** This can be done without breaking
+   compatibility, as the new type will be unused for now.
+
+1. Make `id` in `attachment_asset` mandatory.
+
+    **Migration concerns:** this would need to be done in three steps:
+    1. Update travel advice publisher and manuals publisher to set an arbitrary ID for attachments.
+    2. Republish all travel advice pages and manual sections.
+    3. Make the field mandatory.
+
+1. Add a *mandatory* `details.attachments` field to all formats which
+   can have attachments.
+
+    **Migration concerns:** this would need to be done in four steps:
+    1. Add the field as optional.
+    2. Update publishing apps to write to it.
+    3. Republish all affected documents.
+    4. Make the field mandatory.
+
+1. Add a *mandatory* `details.document_attachments` list to
```

Comment:
I think there's some potential for confusion here between `document_attachments` and `attachments`, as they superficially imply the same thing. To some extent we could resolve this with a well-written description in the schema. Did we consider `featured_attachments` as a name here?

Reply:
I think `featured_attachments` would be a good name for it

Reply:
@kevindew?